### PR TITLE
Drop virtctl expand functests and refactor unit tests 

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

These tests do not require a working cluster and do not provide
additional value to unit tests. Therefore these tests are dropped in
favor of the unit tests in pkg/virtctl/vm/expand_test.go.

The virtctl expand tests are refactored and add test cases for tests removed
in the previous commit are added.

The goal of this and further PRs is to reduce the amount of unnecessary uses of
virtctl in the e2e tests to stabilize tests, to reduce the number of flakes and to
remove redudant tests.

Before this PR:

The same virtctl expand tests are run in unit and functests.

After this PR:

virtctl expand tests run only in unit tests.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer

Merge after https://github.com/kubevirt/kubevirt/pull/12634 was merged

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/virtctl/vm/expand_test.go [hunk 1]
```diff
@@ -21,200 +21,177 @@ package vm_test
...
 
-			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", fileInput, file.Name())
-			err := cmd()
+		BeforeEach(func() {
+			expandSpecInterface = kubecli.NewMockExpandSpecInterface(ctrl)
+			vm = kubecli.NewMinimalVM(vmName)
 
-			Expect(err).To(HaveOccurred())
-			Expect(err).Should(MatchError("error expanding VirtualMachine - testvm in namespace - default: error expanding vm from file"))
+			var err error
```
Comment (jcanocan) [inline pkg/virtctl/vm/expand_test.go:106]: nit: move to the top. Like:

```go
BeforeEach(func() {
  var err error
  expandSpecInterface = kubecli.NewMockExpandSpecInterface(ctrl)
  vm = kubecli.NewMinimalVM(vmName)
[...]
```

Comment (0xFelix) [reply pkg/virtctl/vm/expand_test.go:106]: I wanted to keep it in the place where it is needed. (which is the next line)

**Pull request merged**